src/displayapp/screens/Weather.cpp

```diff
@@ -0,0 +1,203 @@
+  lv_obj_set_style_local_text_color(forecast, LV_TABLE_PART_CELL6, LV_STATE_DEFAULT, Colors::deepOrange);
+
+  lv_obj_align(forecast, nullptr, LV_ALIGN_IN_BOTTOM_LEFT, 0, 0);
+
+  for (int i = 0; i < Controllers::SimpleWeatherService::MaxNbForecastDays; i++) {
+    lv_table_set_col_width(forecast, i, 48);
+    lv_table_set_cell_type(forecast, 1, i, LV_TABLE_PART_CELL2);
+    lv_table_set_cell_align(forecast, 0, i, LV_LABEL_ALIGN_CENTER);
+    lv_table_set_cell_align(forecast, 1, i, LV_LABEL_ALIGN_CENTER);
+    lv_table_set_cell_align(forecast, 2, i, LV_LABEL_ALIGN_CENTER);
+    lv_table_set_cell_align(forecast, 3, i, LV_LABEL_ALIGN_CENTER);
+  }
+
+  taskRefresh = lv_task_create(RefreshTaskCallback, 1000, LV_TASK_PRIO_MID, this);
+  Refresh();
+}
+
+Weather::~Weather() {
+  lv_task_del(taskRefresh);
+  lv_obj_clean(lv_scr_act());
+}
+
+void Weather::Refresh() {
+  currentWeather = weatherService.Current();
+  if (currentWeather.IsUpdated()) {
+    auto optCurrentWeather = currentWeather.Get();
+    if (optCurrentWeather) {
+      int16_t temp = optCurrentWeather->temperature;
+      int16_t minTemp = optCurrentWeather->minTemperature;
+      int16_t maxTemp = optCurrentWeather->maxTemperature;
+      lv_obj_set_style_local_text_color(temperature, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, temperatureColor(temp));
+      char tempUnit = 'C';
+      if (settingsController.GetWeatherFormat() == Controllers::Settings::WeatherFormat::Imperial) {
+        temp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(temp);
+        minTemp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(minTemp);
+        maxTemp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(maxTemp);
+        tempUnit = 'F';
+      }
+      temp = temp / 100 + (temp % 100 >= 50 ? 1 : 0);
+      maxTemp = maxTemp / 100 + (maxTemp % 100 >= 50 ? 1 : 0);
+      minTemp = minTemp / 100 + (minTemp % 100 >= 50 ? 1 : 0);
+      lv_label_set_text(icon, Symbols::GetSymbol(optCurrentWeather->iconId));
+      lv_label_set_text(condition, Symbols::GetCondition(optCurrentWeather->iconId));
+      lv_label_set_text_fmt(temperature, "%d°%c", temp, tempUnit);
+      lv_label_set_text_fmt(minTemperature, "%d°", minTemp);
+      lv_label_set_text_fmt(maxTemperature, "%d°", maxTemp);
+    } else {
+      lv_label_set_text(icon, "");
+      lv_label_set_text(condition, "");
+      lv_label_set_text(temperature, "---");
+      lv_obj_set_style_local_text_color(temperature, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+      lv_label_set_text(minTemperature, "");
+      lv_label_set_text(maxTemperature, "");
+    }
+  }
+
+  currentForecast = weatherService.GetForecast();
+  if (currentForecast.IsUpdated()) {
+    auto optCurrentForecast = currentForecast.Get();
+    if (optCurrentForecast) {
+      std::tm localTime = *std::localtime((const time_t*) &optCurrentForecast->timestamp);
```

Comment:
This cast should be done with `reinterpret_cast`.

```suggestion
      std::tm localTime = *std::localtime(reinterpret_cast<const time_t*>(&optCurrentForecast->timestamp));
```

Reply:
Fixed